xGame/client/src/Components/Match/MatchPlayerStats.js

```diff
@@ -0,0 +1,94 @@
+import React, { useState } from "react";
+import Grid from "@material-ui/core/Grid";
+import PropTypes from "prop-types";
+import "./MatchStyle.css";
+import {
+  getPieceList,
+  showTable,
+  createBoardPiece,
+} from "./HelpfulMatchTools.js";
+
+const MatchPlayerStats = (props) => {
+  const playerName = props.playerName;
+  const activeColor = props.activeColor;
+  const turnCounter = props.turnCounter;
+  const boardState = props.boardState;
+  const opponent = props.opponent;
+  const [takenPieces, setTakenPieces] = useState([]);
+  const [changed, setChanged] = useState(true);
+  const [pieceDisplay, setPieceDisplay] = useState([]);
+
+  const determineTaken = () => {
+    let totalPieces = "";
+    if (opponent) {
+      totalPieces =
+        activeColor === "white" ? getPieceList("white") : getPieceList("black");
+    } else {
+      totalPieces =
+        activeColor === "white" ? getPieceList("black") : getPieceList("white");
+    }
+    for (let piece = 0; piece < boardState.length; piece++) {
+      let indexFound = -1;
+      for (let i = 0; i < totalPieces.length; i++) {
+        if (boardState[piece].pieceName === totalPieces[i]) {
+          indexFound = i;
+        }
+      }
+      if (indexFound >= 0) {
+        totalPieces.splice(indexFound, 1);
+      }
+    }
+    setTakenPieces(totalPieces);
+  };
+
+  const generateDisplay = () => {
+    let tempDisplay = [[]];
+    let rowCount = 0;
+    for (let i = 0; i < takenPieces.length; i++) {
+      tempDisplay[rowCount].push(
+        createBoardPiece(
+          takenPieces[i],
+          "hidden",
+          activeColor + "Captured" + i,
+          null
+        )
+      );
+      if (i % 5 === 0) {
+        tempDisplay.push([]);
+        rowCount++;
+      }
+    }
+    return tempDisplay;
+  };
+
+  const displayTurnCounter = () => {
+    if (!opponent) {
+      return <p>Turn Number {turnCounter}</p>;
+    } else {
+      return <p></p>;
+    }
+  };
+
+  if (changed) {
```

Comment:
This if statement will only run once, because then "changed" will always be set to false. If you want to run "determineTaken()" everytime the boardState changes, I'd suggest replacing this if statement with useEffect like so:
useEffect(() => {
    determineTaken();
}, [boardState]);

Reply:
I'm not super clear what this exactly does, but it works so I changed it